hearpreprocess/pipeline.py

```diff
@@ -594,9 +595,21 @@ def run(self):
         metadata = self.postprocess_all_metadata(metadata)
         _diagnose_split_labels(self.longname, "postprocessed", metadata)
 
-        # Split the metadata to create valid and test set from train if they are not
-        # created explicitly in get_all_metadata
-        metadata = self.split_train_test_val(metadata)
+        # If explicit folds are not defined, `split_train_test_val` will
+        # be called, to ensure all the splits, i.e. train, test and valid
+        # are present (by sampling )
+        if "folds" not in self.task_config:
+            assert set(self.task_config["splits"]) == set(SPLITS), "If folds are not "
+            "defined, the `splits` key in the task configuration should "
+            "be SPLITS i.e. ['train', 'test', 'valid']"
```

Comment:
Are we introducing the "splits" key in this PR? -- and this is set in the runner?